pygmt/base_plotting.py

```diff
@@ -1922,3 +1922,248 @@ def update_pointers(data_pointers):
+            (``scaling='w'``). Can be combined with the confidence variable.
+
+        uncertainty_color : str
+            Sets the color or shade used for filling uncertainty wedges
+            (``scaling='w'``) or velocity error ellipses (``scaling='e'`` or
+            ``scaling='r'``). [If *uncertainty_color* is not specified, the
+            uncertainty regions will be transparent]. More details on
+            :gmt-docs:`cookbook/features.html#gfill-attrib`.
+
+        facecolor : str
+            Select color or pattern for filling of symbols or polygons
+            [Default is no fill]. More details on
+            :gmt-docs:cookbook/features.html#gfill-attrib`.
+
+        line: str
+            Draw lines. Ellipses and fault planes will have their outlines
+            drawn using current pen (see *pen*).
+
+        no_clip: str
+            Do NOT skip symbols that fall outside the frame boundary specified
+            by *region*. [Default plots symbols inside frame only].
+
+        pen : str
+            Set pen attributes for velocity arrows, ellipse circumference and
+            fault plane edges. [Defaults: width = default, color = black,
+            style = solid].
+
+        {U}
+        {V}
+        {XY}
+        {p}
+        {t}
+        """
+        kwargs = self._preprocess(**kwargs)
+
+        kind = data_kind(data)
+
+        extra_arrays = []
+        if "G" in kwargs and not isinstance(kwargs["G"], str):
+            if kind != "vectors":
+                raise GMTInvalidInput(
+                    "Can't use arrays for color if data is matrix or file."
+                )
+            extra_arrays.append(kwargs["G"])
+            del kwargs["G"]
+
+        if scaling is not None:
+            if not isinstance(scaling,str):
+                raise GMTInvalidInput(
+                    "Scaling is a requirred argument and has to be a string."
+                )
+            extra_arrays.append(scaling)
+
+        with Session() as lib:
+            # Choose how data will be passed in to the module
+            if kind == "file":
+                file_context = dummy_context(data)
+            elif kind == "matrix":
+                if pd.api.types.is_numeric_dtype(data):
+                    file_context = lib.virtualfile_from_matrix(data)
+                elif isinstance(data,np.ndarray):
+                    raise GMTInvalidInput(
+                        "Text columns are not supported with numpy array. "
+                        "They are only supported with file or "
+                        "pandas dataframe inputs."
+                    )
+                elif isinstance(data,pd.core.frame.DataFrame):
+                    file_context = lib.virtualfile_from_vectors(
+                        *[data[column] for column in data]
+                    )
+                else:
+                    raise GMTInvalidInput("Unrecognized data type: {}".format(type(data)))
```

Comment:
```suggestion
                    raise GMTInvalidInput(f"Unrecognized data type: {type(data)}")
```
Possible to use an [f-string](https://realpython.com/python-f-strings/) here.